### ISSUE TITLE
Tags work funny on the GUI

### ISSUE THREAD

**Issue by Szelethus** [Ericsson/codechecker#3889]
**Describe the bug**
Observe the following inconsitencies:
```c++
// test1.cpp
#include "stdlib.h"

void a() {
  int i = 0;
  (void)(10 / i);
}

void b() {
  int *i = 0;
  *i = 5;
}

void c() {
  int i = 0;
  //
  i = 5;
}

// void d() {
//   int *k = new int;
//   (void)k;
// }

void e() {
  int *k = new int;
  // codechecker_suppress [all] SUPPRESS ALL
  free(k);
}

void f() {
  int *k = new int;
  delete k;
  *k = 5;
}
```

I'm analyzing this file into the results folder `result1/` and store it into `SingleRun` **with the tag `tag1`**:
```
CodeChecker check -b "g++ -c test1.cpp" -o result1 --analyzers clangsa -c
CodeChecker store result1/ -n "SingleRun" --tag tag1 --url 0.0.0.0/5569
```
In the GUI, in the reports tab, after pressing "CLEAR ALL FILTERS", I see that I have 5 reports:
![image](https://user-images.githubusercontent.com/23276031/233091177-5dadb3cf-e6eb-46f2-acfa-e579db80277e.png)

After choosing the tag under "Run / Tag Filter", I can only see 4:
![image](https://user-images.githubusercontent.com/23276031/233091460-0cf1d978-64fe-4928-8a1c-00dbbc990b0c.png)

Mind that the report that was suppressed via a source code comment is now missing. If I set a false positive review status rule for one of the reports, that one won't be missing, interestingly.

**CodeChecker version**
```
---------------------------------------------------------------
Kind                 | Version                                 
---------------------------------------------------------------
Base package version | 6.22.0                                  
Package build date   | 2023-04-18T10:11                        
Git commit ID (hash) | 0c86979fec4dbcc3b2aa24c766f9a08a28f78fcb
Git tag information  | 6.22                                    
---------------------------------------------------------------

[INFO 2023-04-19 15:13] - CodeChecker web version:
------------------------------------------------------------------------------
Kind                                | Version                                 
------------------------------------------------------------------------------
Base package version                | 6.22.0                                  
Package build date                  | 2023-04-18T10:11                        
Git commit ID (hash)                | 0c86979fec4dbcc3b2aa24c766f9a08a28f78fcb
Git tag information                 | 6.22                                    
Server supported Thrift API version | 6.53                                    
Client Thrift API version           | 6.53                                    
------------------------------------------------------------------------------
```

**Comment by Szelethus** [comment]
I managed to observe this behaviour on the cmd as well. Note that I used `--tag` in the second command.
```bash
$ CodeChecker cmd results SingleRun --review-status --detection-status  --url 0.0.0.0/Tag
(displays 5 reports)
$ CodeChecker cmd results SingleRun --tag tag1 --review-status --detection-status  --url 0.0.0.0/Tag
(displays 4 reports)
```

**Comment by Szelethus** [comment]
Actually, this seems to be a feature: https://github.com/Ericsson/codechecker/pull/1269

**Comment by bruntib** [comment]
I would close this ticket, since it is the expected behavior for CodeChecker. If I'm missing something, then feel free to reopen it.
